Source/GSXib5KeyedUnarchiver.m

```diff
@@ -3201,6 +3290,42 @@ - (BOOL) decodeBoolForKey: (NSString *)key
   return flag;
 }
 
+- (NSInteger) decodeIntegerForKey: (NSString *)key
```

Comment:
There should be no reason for this method. Please try to remove it and test again. The code in the super implementation and in decodeObjectForKey: should do the trick.
If there is still an issue it might be the order of the checks in the decodeObjectForKey: method.

   `if ([XmlKeyMapTable objectForKey: key])
        {
          object = [self decodeObjectForKey: [XmlKeyMapTable objectForKey: key]];
        }
      else if ([XmlKeyToDecoderSelectorMap objectForKey: key])
        {`

You could try to change that order, in this case.

Reply:
ok

Reply:
It's not decoding an object... it's decoding an integer.

Reply:
Did you actually look at the implementation of the other int decoding methods in GSXibKeyedUnarchiver.m?

`- (int64_t) decodeInt64ForKey: (NSString*)aKey
{
  id o = [self decodeObjectForKey: aKey];

  if (o != nil)
    {
      if (([o isKindOfClass: [NSNumber class]] == YES) ||
          ([o isKindOfClass: [NSString class]] == YES))
	{
	  long long l = [o longLongValue];

	  return l;
	}
      else
	{
	  [NSException raise: NSInvalidUnarchiveOperationException
		      format: @"[%@ +%@]: value for key(%@) is '%@'",
	    NSStringFromClass([self class]), NSStringFromSelector(_cmd),
	    aKey, o];
	}
    }
  return 0;
}`

Reply:
> Did you actually look at the implementation of the other int decoding methods in GSXibKeyedUnarchiver.m?
> 
> `- (int64_t) decodeInt64ForKey: (NSString*)aKey
> {
> id o = [self decodeObjectForKey: aKey];
> 
> if (o != nil)
> {
> if (([o isKindOfClass: [NSNumber class]] == YES) ||
> ([o isKindOfClass: [NSString class]] == YES))
> {
> long long l = [o longLongValue];
> 
> ```
>   return l;
> }
>   else
> {
>   [NSException raise: NSInvalidUnarchiveOperationException
> 	      format: @"[%@ +%@]: value for key(%@) is '%@'",
>     NSStringFromClass([self class]), NSStringFromSelector(_cmd),
>     aKey, o];
> }
> }
> ```
> 
> return 0;
> }`

Yes, I looked.  Please look at the .xib vs. the .nib?   We are going from a text-based value like "none" to an integer which is mapped via an enumerated value to something like NS....None which is 0.   This is not as simple as going from a string value which is a number to an integer.   If it were the existing method would be sufficient.   I am mapping from a text-word to an integer-value.    This is similar to the reason why decodeFromBool exists in GSXib5KeyedUnarchiver.

Reply:
The method you copied above does NOT cover the use case I am trying to implement.

Reply:
> Did you actually look at the implementation of the other int decoding methods in GSXibKeyedUnarchiver.m?
> 
> `- (int64_t) decodeInt64ForKey: (NSString*)aKey
> {
> id o = [self decodeObjectForKey: aKey];
> 
> if (o != nil)
> {
> if (([o isKindOfClass: [NSNumber class]] == YES) ||
> ([o isKindOfClass: [NSString class]] == YES))
> {
> long long l = [o longLongValue];
> 
> ```
>   return l;
> }
>   else
> {
>   [NSException raise: NSInvalidUnarchiveOperationException
> 	      format: @"[%@ +%@]: value for key(%@) is '%@'",
>     NSStringFromClass([self class]), NSStringFromSelector(_cmd),
>     aKey, o];
> }
> }
> ```
> 
> return 0;
> }`

I see it.  I will remove the method and try again.

Reply:
Changed to use decodeIntForKey: this works.  Thanks for the feedback.

Reply:
This has been done.